tests/console/validate_encrypt.pm

```diff
@@ -19,30 +19,72 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
 
-sub run {
+sub parse_crypttab {
+    my $path = shift;
+    return {split(/\s+/, script_output("cat $path"))};
+}
 
-    record_info('crypttab file', 'Verify whether \'/etc/crypttab\' (encrypted device table) file exists.');
-    assert_script_run('test -f /etc/crypttab', fail_message => 'No /etc/crypttab found!');
+sub parse_cryptsetup_status {
+    my $dev    = shift;
+    my @lines  = split(/\n/, script_output("cryptsetup status $dev"));
+    my %status = ();
+    $status{message} = shift @lines;
+    foreach (@lines) {
+        if ($_ =~ /\s+(.*):\s+(.*)$/) {
+            my ($k, $v) = ($1, $2);
+            $k =~ s/ /_/g;
+            $status{$k} = $v;
+        }
+    }
+    return \%status;
+}
 
-    record_info('crypt vol status', 'Verify the crypted volumes are active');
-    my @crypted_volumes = script_output q[cat /etc/crypttab | awk '{print $1}'];
-    foreach (@crypted_volumes) {
-        assert_script_run qq[cryptsetup status $_ | grep "is active"];
+sub verify_crypttab {
+    my $crypttab = shift;
+    record_info("crypttab", "Verify file existence and content");
+    assert_script_run("test -f $crypttab->{path}", fail_message => "No $crypttab->{path} found");
+    my $content_current = parse_crypttab($crypttab->{path});
+    foreach (@{$crypttab->{devices}}) {
+        die "Device $_ not found in $crypttab->{path}" unless exists($content_current->{$_});
     }
+}
 
-    record_info('Verify LUKS', 'Verify binary backups of LUKS header and keyslot areas storing and restoring');
-    foreach (@crypted_volumes) {
-        my $bkp_file = '/root/bkp_luks_header';
-
-        my $device = script_output q[cat /etc/crypttab | awk '{print $2}'];
-        next if (script_run('cryptsetup -v isLuks ' . $device) != 0);
-        assert_script_run('cryptsetup -v luksUUID ' . $device);
-        assert_script_run('cryptsetup -v luksDump ' . $device);
-        assert_script_run('cryptsetup -v luksHeaderBackup ' . $device . ' --header-backup-file ' . $bkp_file);
-        validate_script_output("file $bkp_file", sub { m/\bLUKS\sencrypted\sfile\b/ });
-        assert_script_run('cryptsetup -v --batch-mode luksHeaderRestore ' . $device . ' --header-backup-file ' . $bkp_file);
+sub verify_cryptsetup_status {
+    my $status = shift;
+    record_info("active volumes", "Verify crypted volumes are active");
+    foreach my $dev (keys %{$status}) {
+        my $status_current = parse_cryptsetup_status($dev);
+        foreach my $property (keys %{$status->{$dev}}) {
```

Comment:
I would add one more check, that set of key matches, as we might miss the bug if e.g. there is `cr_vda5` in SUT.

Reply:
Done at the beginning of the test on validation of `/etc/crypttab`. I try to play with `Test::Assert` for comparing lists before this code, but the code turned to be pretty ugly, one-liner but no really clear error. I'm not sure if looking for exact match is a good philosophy but it doesn't hurt. It adds additional extra intelligence to the test. It is a really good point that you mentioned, made me think too much :)